Applications/Calculator/Calculator.h

```diff
@@ -0,0 +1,46 @@
+#pragma once
+
+// This type implements the regular calculator
+// behavior, such as performing arithmetic
+// operations and providing a memory cell.
+// It does not deal with number input; you
+// have to pass in already parsed double
+// values.
+
+class Calculator final {
```

Comment:
Note that `final` is usually used for leaf classes in some inheritance hierarchy to denote that the inheritance stops here.
It allows the compiler to do things like devirtualizing calls to virtual functions on a pointer-to-leaf-class.
Of course, it's not a bad thing to use it like this either, but all it communicates is "give me a compiler error if I try to inherit from `Calculator`" :)

Reply:
> Note that `final` is usually used for leaf classes in some inheritance hierarchy to denote that the inheritance stops here. It allows the compiler to do things like devirtualizing calls to virtual functions on a pointer-to-leaf-class.

I know! The reason I'm using `final` here is that I'm used (from the Java world) to following the principle that states that you should either design your class for inheritance or make it `final`. Of course you can chose not to follow this principle in Serenity, and anyway this is an app class nobody's going to try to subclass, so it doesn't matter. Removing `final` then.

Reply:
Nah we can leave it in, I like the idea! I just never saw it before :)

Reply:
Okay then, left as is.